bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -120,16 +124,32 @@ public Response create(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value
         }
 
         // turn the ThingDTO's configuration into a Configuration
-        Configuration configuration = getConfiguration(thingBean);
+        Configuration configuration = convertConfiguration(thingBean.configuration);
 
         Status status;
         Thing thing = thingRegistry.get(thingUIDObject);
 
         // does the Thing already exist?
-        if (null == thing) {
-            // if not, create new Thing
+        if (thing == null) {
+        	
+            // if not, create new Channels and add them to a new Thing
+            List<Channel> channels = new ArrayList<Channel>();
+            for (ChannelDTO channelBean : thingBean.channels) {
+            	
+            	String channelId = channelBean.id;
+            	if (channelId == null) {
```

Comment:
Is this at all a valid situation? I would assume that a client has to provide an id here.
Note: For rule modules within a rule, we expect the same, so requiring an id here would be in line with that.